src/TrackerSecondary.java

```diff
@@ -0,0 +1,278 @@
+     * @requires |this| > 0
+     * @ensures <pre>
+     * highest is in this  and
+     * all name.value in DOMAIN(highest) >= all name.value in DOMAIN(this)
+     * </pre>
+     */
+    @Override
+    public Tracker highest() {
+
+        //set maxValue to any value from the Tracker
+        Tracker result = this.newInstance();
+        Map.Pair<String, Integer> any = this.removeAny();
+        int maxValue = any.value();
+        this.setValue(any.key(), any.value());
+
+        //Find the highest value
+        for (Map.Pair<String, Integer> item : this) {
+
+            if (item.value() > maxValue) {
+                maxValue = item.value();
+            }
+
+            this.setValue(item.key(), item.value());
+        }
+
+        //Add all pairs with the highest value to result
+        for (Map.Pair<String, Integer> item : this) {
+
+            if (item.value() >= maxValue) {
+                result.setValue(item.key(), item.value());
+            }
+
+            this.setValue(item.key(), item.value());
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns all of the pairs with the lowest value.
+     *
+     * @return pairs with lowest value
+     * @requires |this| > 0
+     * @ensures <pre>
+     * lowest is in this  and
+     * all name.value in DOMAIN(lowest) >= all name.value in DOMAIN(this)
+     * </pre>
+     */
+    @Override
+    public Tracker lowest() {
+
+        //set minValue to any value from the Tracker
+        Tracker result = this.newInstance();
+        Map.Pair<String, Integer> any = this.removeAny();
+        int minValue = any.value();
+        this.setValue(any.key(), any.value());
+
+        //Find the highest value
+        for (Map.Pair<String, Integer> item : this) {
+
+            if (item.value() < minValue) {
+                minValue = item.value();
+            }
+
+            this.setValue(item.key(), item.value());
+        }
+
+        //Add all pairs with the highest value to result
+        for (Map.Pair<String, Integer> item : this) {
+
+            if (item.value() <= minValue) {
+                result.setValue(item.key(), item.value());
+            }
+
+            this.setValue(item.key(), item.value());
+        }
+
+        return result;
+
+    }
+
+    /**
+     * Reports the value of {@code this} as a String.
+     *
+     * @return a String representation of {@code this}
+     */
+    @Override
+    public String toString() {
+
+        String result = "{";
+
+        for (Map.Pair<String, Integer> item : this) {
+
+            result += "(" + item.key() + ", " + item.value().toString() + "), ";
+        }
+
+        result += "}";
+
+        return result;
+    }
+
+    /**
+     * Reports wheather {@code this} and {@code t} are equal.
+     *
+     * @param t
+     *            the {@code Tracker} to compare to {@code this}
+     * @return wheather {@code this} and {@code t} are equal
+     */
+    public boolean equals(Tracker t) {
```

Comment:
See the common methods slides. There is a more "idiomatic" way of solving this problem.